File: src/Addins/Eto.Forms.Templates/content/EtoApp-CSharp/EtoApp/MainForm.jeto.cs
#if (UseJeto)
using System;
using System.Collections.Generic;
using Eto.Forms;
using Eto.Drawing;
using Eto.Serialization.Json;

namespace EtoApp
{	
	public class MainForm : Form
	{	
		public MainForm()
		{
			JsonReader.Load(this);
		}

		protected void HandleClickMe(object sender, EventArgs e)
		{
			MessageBox.Show("I was clicked!");
		}

		protected void HandleQuit(object sender, EventArgs e)
		{
			Application.Instance.Quit();
		}
	}
}
#endif

File: src/Addins/Eto.Forms.Templates/content/App-FSharp/EtoApp.1/MainForm.jeto.fs
#if (UseJeto)
namespace EtoApp._1

open System
open Eto.Forms
open Eto.Drawing
open Eto.Serialization.Json;

type MainForm() as this = 
    inherit Form()

    do 
        JsonReader.Load(this)

#if IsForm
    member this.HandleClickMe(sender:obj, e:EventArgs) =
        MessageBox.Show("Hello!") |> ignore

    member this.HandleAbout(sender:obj, e:EventArgs) =
        let dlg = new AboutDialog()
        dlg.ShowDialog(this) |> ignore

    member this.HandleQuit(sender:obj, e:EventArgs) =
        Application.Instance.Quit()
#endif
#endif